### ISSUE TITLE
Question about common failure cases

### ISSUE THREAD

**Issue by tcm03** [URDF-Anything-plus/Code#1]
Hi,

First of all, thanks for the great work that continues from URDF-Anything and for publishing it early to the open community.
I've run some tests on the provided model checkpoints with **In-The-Wild** settings and I observed some problems with the cases below.

For the first example of a real-life faucet suggested from the repo, I can see that it generates good meshes with correct articulation:

<img width="1024" height="852" alt="Image" src="https://github.com/user-attachments/assets/3cd593c6-0dd6-4a32-a8a5-1f3471eb63a6" />

<img width="422" height="451" alt="Image" src="https://github.com/user-attachments/assets/12f5b949-51b7-434e-8489-8152c00c96c6" />

But when I tested with another real object, a cabinet in the office, it seems that the model fails to generate high-quality meshes both for the base and the drawer, and the articulation type is revolute instead of prismatic:

<img width="640" height="853" alt="Image" src="https://github.com/user-attachments/assets/b0110dbd-ee3d-4e5f-8d1d-19e94e4d5661" />

<img width="1920" height="689" alt="Image" src="https://github.com/user-attachments/assets/5a737328-3e7c-4fce-8dc6-03eeb2867cff" />

<img width="1024" height="332" alt="Image" src="https://github.com/user-attachments/assets/e6204a67-9ea7-41dd-b4be-4b2135e1be12" />

I further tested with some synthetic objects from the PartNet-Mobility dataset, such as a big storage cabinet (id 44826)

<img width="640" height="480" alt="Image" src="https://github.com/user-attachments/assets/7c1c2d38-d4bf-4a00-8ecd-b88bdf73a049" />

<img width="640" height="377" alt="Image" src="https://github.com/user-attachments/assets/d949b8e7-7811-49c2-81c7-ad7d4e32fc12" />

or a laptop (id 10213)

<img width="640" height="480" alt="Image" src="https://github.com/user-attachments/assets/1b506dd0-7407-4599-a0a6-de105770861e" />

<img width="1473" height="789" alt="Image" src="https://github.com/user-attachments/assets/35bf5461-e109-4890-920c-3f7cc395981b" />

<img width="1024" height="317" alt="Image" src="https://github.com/user-attachments/assets/3858f21e-2559-4cc8-9361-449784ec2f0d" />

As can be seen, it seems that the model fails for these cases in terms of geometry and articulation (the last case is wrong axis direction for revolute).

Since some cases I experimented with above are basic objects, and I have used all the provided checkpoints for TripoSG, RMBG, DINO, and epoch_200.pth, I wonder if there is anything that could be wrong in my setup that causes unexpected errors as above? Otherwise, I would love to hear more from the authors

Thanks in advance.

**Comment by ZhuangzheWu** [comment]
You can try using this image of 10213:

<img width="348" height="267" alt="Image" src="https://github.com/user-attachments/assets/b6fbf43a-1253-48ed-92c4-ad8f8dbcbb90" />

and here is the result of in-the-wild mode:

<img width="400" height="300" alt="Image" src="https://github.com/user-attachments/assets/a837862c-abbd-45ec-b860-d0f49df472ed" />

All 3D meshes in our training set are sourced from PartNet and PartNet-Mobility. They have fixed orientations and poses, and their structures are complete. In the in-the-wild setting, TripoSG may produce outputs with less ideal poses or quality, which are unfamiliar to the model, so performance differences may arise. This is why we separate the results into in-domain (test set) results and in-the-wild results.

This is our preliminary version—you can try the example in our inference.sh. We will continue to improve the model’s capabilities in the future, potentially leveraging 3D foundation models that are more powerful than TripoSG. Thank you for your interest and questions!
